core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -253,26 +251,8 @@ private void deliver() {
        * be in unprocessed.
        */
       boolean stalled = unprocessed.readableBytes() == 0;
-
-      if (endOfStream && stalled) {
-        boolean havePartialMessage = nextFrame != null && nextFrame.readableBytes() > 0;
```

Comment:
It being missed on clients is an oversight (probably lost in an earlier refactoring). I've filed https://github.com/grpc/grpc-java/issues/3264